core/trino-main/src/main/java/io/trino/operator/exchange/UniformPartitionRebalancer.java

```diff
@@ -162,17 +162,15 @@ private synchronized void rebalancePartitions(List<Long> writerPhysicalWrittenBy
             for (WriterId minSkewedWriter : minSkewedWriters) {
                 // There's no need to add the maxWriter back to priority queues if no partition rebalancing happened
                 if (context.rebalancePartition(maxWriter, minSkewedWriter)) {
-                    maxWriters.addOrUpdate(maxWriter, context.getWriterEstimatedWrittenBytes(maxWriter));
-                    minWriters.addOrUpdate(maxWriter, Long.MAX_VALUE - context.getWriterEstimatedWrittenBytes(maxWriter));
+                    for (Integer writer: context.rebalancedAffectedWriters) {
```

Comment:
Please make it a getter `context.getRebalancedAffectedWriters()`

Reply:
> Do you mean update all the writers although some of them have no change?
Yes

> more compution spending
I don't think it matters a lot here since `UniformPartitionRebalancer#rebalancePartitions(..)` is only called after every 50MBs has been written, so it's not that frequent. Additionally, in most cases, the total number of writers will be 32 or a little more.

Reply:
Why did you remove this? @zombee0